### PR TITLE
chore: mark internal used props as deprecated

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

Mark internal used props as deprecated. BaseXXXProps(e.g., `BaseButtonProps`) are intended for internal use only. External users should go ahead use `ButtonProps` for instance.

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/tests/__snapshots__/props-table.test.ts.snap [hunk 1]
```diff
@@ -11556,7 +11556,7 @@ exports[`Props Table 1`] = `
                     \\"BaseTextFieldProps\\": {
                         \\"name\\": \\"BaseTextFieldProps\\",
                         \\"type\\": \\"extends TextFieldOptions, BaseInputProps\\",
-                        \\"description\\": \\"\\",
+                        \\"description\\": \\"0: / 1: * 2: * 3:   4: @ 5: d 6: e 7: p 8: r 9: e 10: c 11: a 12: t 13: e 14: d 15:   16: F 17: o 18: r 19:   20: i 21: n 22: t 23: e 24: r 25: n 26: a 27: l 28:   29: u 30: s 31: e 32:   33: o 34: n 35: l 36: y 37:   38: * 39: /\\",
```
Comment (dbanksdesign) [inline docs/tests/__snapshots__/props-table.test.ts.snap:11559]: This looks like the new jsdoc comment is getting split into an array? Have we verified that this doesn't look weird on the docs site? Do we show these types in props tables?

Comment (calebpollman) [reply docs/tests/__snapshots__/props-table.test.ts.snap:11559]: +1. Don't have a ton of understanding on how the props tables work but it does seem like these types should be omitted from it?

Comment (zchenwei) [reply docs/tests/__snapshots__/props-table.test.ts.snap:11559]: The script needs to update after `TextArea` is separated from `TextField`. Will update

File: docs/tests/__snapshots__/props-table.test.ts.snap [hunk 2]
```diff
@@ -11552,55 +11552,6 @@ exports[`Props Table 1`] = `
...
                         \\"isOptional\\": true
-                    },
-                    \\"BaseTextFieldProps\\": {
-                        \\"name\\": \\"BaseTextFieldProps\\",
-                        \\"type\\": \\"extends TextFieldOptions, BaseInputProps\\",
-                        \\"description\\": \\"\\",
-                        \\"category\\": \\"TextField\\",
-                        \\"isOptional\\": true
-                    },
-                    \\"resize\\": {
```
Comment (zchenwei) [inline docs/tests/__snapshots__/props-table.test.ts.snap:11563]: This prop and a few others are from `TextArea`. It should not be included in `TextField` anymore.

File: docs/tests/__snapshots__/props-table.test.ts.snap [hunk 3]
```diff
@@ -12371,13 +12322,6 @@ exports[`Props Table 1`] = `
                         \\"category\\": \\"BaseComponentProps\\",
                         \\"isOptional\\": true
                     },
-                    \\"ForwardRefPrimitive\\": {
```
Comment (zchenwei) [inline docs/tests/__snapshots__/props-table.test.ts.snap:12374]: This is an internal interface. Not a prop on View

**Review by dbanksdesign — APPROVED** [review]
LGTM!

**Review by calebpollman — APPROVED** [review]

**Pull request merged**